ledger/participant-integration-api/src/main/scala/platform/indexer/RecoveringIndexer.scala

```diff
@@ -79,10 +79,11 @@ private[indexer] final class RecoveringIndexer(
             logger.info("Restarting Indexer Server")
             val newSubscription = subscribe()
             if (subscription.compareAndSet(oldSubscription, newSubscription)) {
-              resubscribeOnFailure(newSubscription)
               newSubscription.asFuture.map { _ =>
                 logger.info("Restarted Indexer Server")
               }
+              resubscribeOnFailure(newSubscription)
```

Comment:
These two things will run in parallel, so I don't think this will help flakiness.

Reply:
Only if you use `parasitic` (or `DirectExecutionContext`) everywhere!